src/Transformer/EnumTransformerFactory.php

```diff
@@ -38,12 +38,14 @@ private function isEnumType(Type $type, bool $backed = false): bool
             return false;
         }
 
-        if (!is_subclass_of($type->getClassName(), \UnitEnum::class)) {
-            return false;
-        }
+        if (null !== $type->getClassName()) {
```

Comment:
this change is strange... `is_subclass_of()` accepts `mixed`, thus `null`

https://3v4l.org/Av3ia